torch/csrc/jit/script/module.h

```diff
@@ -534,9 +315,17 @@ struct Module {
     auto offset = find_offset(name, EntityType::MODULE);
     return offset ? modules_[*offset] : nullptr;
   }
-  Method* find_method(const std::string& name) {
+  Method* find_method(const std::string& name) const {
     auto offset = find_offset(name, EntityType::METHOD);
-    return offset ? methods_[*offset].get() : nullptr;
+    if (offset) {
+      return methods_[*offset].get();
+    }
+
+    if (Function* fn = class_cu().find_function(name).get()) {
+      return &const_cast<Module*>(this)->lower_first_class_method(fn);
```

Comment:
this const_cast seems suspicious…but I guess this will go away after we remove the lowering.

Reply:
The list of Methods is basically a cache on top of the Module CU. In the future Method will turn into a simple value type that just holds IValue and a pointer to Function. So this const_cast is just so that we can update the cache. It will go away once we no longer have to lower. Technically we should be holding a lock while doing the lowering because this function may be run from multiple threads. I can update that.